### PR TITLE
201-vmss-scale-existing: Updated azuredeploy.parameters.json to accept prereqs outputs

### PR DESCRIPTION
**Pull request by PraveenAnil**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 201-vmss-scale-existing/prereqs/prereq.azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+                        "backendPort": 3389
+                    }
+                }]
+            }
+        },
+        {
+            "type": "Microsoft.Compute/virtualMachineScaleSets",
+            "name": "existingvmss",
+            "location": "[resourceGroup().location]",
+            "apiVersion": "2016-04-30-preview",
```
Comment (bmoore-msft) [inline 201-vmss-scale-existing/prereqs/prereq.azuredeploy.json]: let's update this to 2017-03-30

File: 201-vmss-scale-existing/prereqs/prereq.azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+                    "networkProfile": {
+                        "networkInterfaceConfigurations": [{
+                            "name": "[concat(variables('namingInfix'), 'nic')]",
+                            "properties": {
+                                "primary": true,
+                                "ipConfigurations": [{
+                                    "name": "[concat(variables('namingInfix'), 'ipconfig')]",
+                                    "properties": {
+                                        "subnet": {
+                                            "id": "[concat('/subscriptions/', subscription().subscriptionId,'/resourceGroups/', resourceGroup().name, '/providers/Microsoft.Network/virtualNetworks/', variables('virtualNetworkName'), '/subnets/', variables('subnetName'))]"
```
Comment (bmoore-msft) [inline 201-vmss-scale-existing/prereqs/prereq.azuredeploy.json]: simplify with resourceId('Microsoft.Network/virtualNetworks/subnets, variables('virtualNetworkName'), variables('subnetName'))

File: 201-vmss-scale-existing/prereqs/prereq.azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+                        },
+                        "imageReference": {
+                            "publisher": "MicrosoftWindowsServer",
+                            "offer": "WindowsServer",
+                            "sku": "2012-R2-Datacenter",
+                            "version": "latest"
+                        }
+                    },
+                    "osProfile": {
+                        "computerNamePrefix": "[toLower(substring(concat('existingvmss', uniqueString(resourceGroup().id)), 0, 9))]",
```
Comment (bmoore-msft) [inline 201-vmss-scale-existing/prereqs/prereq.azuredeploy.json]: I think you can just use existingVmss here... no need for uniqueString

**Comment by PraveenAnil** [comment]
Thanks for the feedback. I will make the suggested changes.

**Pull request merged**